docs/how-tos/sensor-module.md

```diff
@@ -0,0 +1,414 @@
+---
+title: "How to create a sensor module with Python"
+linkTitle: "Create a sensor module"
+type: "docs"
+weight: 26
+images: ["/icons/components/sensor.svg"]
+tags: ["modular resources", "components", "services", "registry"]
+description: "Add a custom resource by creating and deploying a module to your machine."
+---
+
+A sensor is anything that collects data.
+
+A sensor could be something we typically think of as a sensor, like a temperature and humidity sensor, or it could be a "virtual," non-hardware sensor like a service that gets stock market data.
+
+Since a sensor can be so many different things, there's a good chance you're on this page because though there are various [built-in and modular sensor models available in Viam](/components/sensor/#available-models), you have a different, unsupported sort of sensor you'd like to use.
+
+Making a module to support your sensor will allow you to use it with Viam's data capture and sync tools, as well as using the sensor API (using any of the different programming language [SDKs](/sdks/)) to get readings from it.
+
+{{% alert title="In this page" color="info" %}}
+
+1. [Start with a test script](#start-with-a-test-script)
+1. [Generate boilerplate module code](#generate-boilerplate-module-code)
+1. [Implement the sensor API](#implement-the-sensor-api)
+1. [Make the module executable](#make-the-module-executable)
+1. [Test your module locally](#test-your-module-locally)
+1. [Upload your module](#upload-your-module-to-the-registry)
+
+{{% /alert %}}
+
+## Start with a test script
+
+Start by getting a test script working so you can check that the sensor code itself works before packaging it into a module.
+
+Since this how-to uses Python, you need a Python test script so that you can more easily wrap it in a Python-based module.
+You'll still be able to use any of Viam's SDKs with the module.
```

Comment:
```suggestion
You'll still be able to use any of Viam's SDKs to query data from machines that use the module.
```

Reply:
Would you call it querying if it's getting readings directly from the machine as opposed to from the database?

Reply:
```suggestion
You'll still be able to use any of Viam's SDKs to get readings from machines that use the module.
```